src/bridge/views/AcceptTransfer.js

```diff
@@ -65,7 +65,7 @@ class AcceptTransfer extends React.Component {
     return Just(azimuth.ecliptic.transferFrom(
       validContracts,
       owner,
-      state.receivingAddress,
+      state.receivingAddress.toLowerCase(),
```

Comment:
idk if this is the right approach. If you enter an address with invalid checksum, the app *should* shout at you... but not in a "crash and burn" kinda way. Probably want to just do checksum checking on the address input component and have it consider input as invalid appropriately?

Reply:
Left a comment about this here, but maybe there's a better solution: https://github.com/urbit/bridge/issues/52#issuecomment-479678341

IMO, the validator function should handle things like address input before we even call azimuth.ecliptic. The exception we get back from azimuth.ecliptic is a big stack trace that's hard to map to a specific validation error:

![image](https://user-images.githubusercontent.com/583272/55840061-f0297880-5ade-11e9-9877-76119d816a22.png)

I'm not sure what exceptional cases that azimuth.ecliptic might throw or where to display them on the page. Maybe we just add a try/catch with a noop for now to reduce development pain?

Reply:
Regardless of what we do about exceptions, I still think lowerCasing is a correct thing to do to fix this specific problem; users should be able to input uppercase addresses and have it go through successfully (right?)

Reply:
I think we're saying the same thing, at least in your first comment!

What I mean to say, is that the component that takes an Ethereum address and updates `receivingAddress` is the one that should do checksum checking, and consider the input invalid if the input is checksummed but not correct, not setting `receivingAddress` in that case.

Reply:
> users should be able to input uppercase addresses and have it go through successfully (right?)

To answer this more directly: the point of addresses with uppercase characters in them is that they're checksummed. If that checksum doesn't check out, we should inform the user that they may have made a typo, or something went wrong during copy-pasting. Just always lowercasing their input defeats the purpose of having checksummed addresses. See also rationale for the standard [here](https://eips.ethereum.org/EIPS/eip-55).

Reply:
Oh, okay, that makes sense! Definitely the wrong solution here then, hah.

Reply:
So we should probably update isValidAddress in lib/wallet's validation logic then, right? https://github.com/urbit/bridge/blob/master/src/bridge/lib/wallet.js#L72

Reply:
That's probably the one, yeah. Except that checksum validation logic lives in web3, and I'm not sure where else we have it accessible, so that may not be trivial to pull into that lib... ):

Reply:
Looks like we've already implemented it right below where it's supposed to be called, haha. https://github.com/urbit/bridge/blob/master/src/bridge/lib/wallet.js#L75

Still trying to understand exactly what we do here

1) User types a lowercase address
   - This is okay, and validation should return true (?)
2) User types a mixed case address
   - Run it through the checksum validator, and if the checksum address matches up, validation should return true
3) User types an uppercase address 
  - This is okay, and validation should return true (?)

Reply:
Oh, sorry for the delay on this.

1 and 2 are correct. 3 should be treated as identical to case 2. "Fully uppercase address" is not a common/valid notation, unless that happens to be the checksum version of the address. (Not that that's ever the case, but "string contains uppercase" is a good trigger for "need to verify checksum".)

Reply:
That's what I landed on. Pushing a version that has this, just confirming it works real quick

```
const isValidAddress = (address) => {
  let validLength = /^0x[0-9a-fA-F]{40}$/.test(address)
  let isLowercase = address === address.toLowerCase()
  let validChecksum = address === toChecksumAddress(address)

  if (isLowercase) return validLength
  else return validChecksum && validLength
}
```

Reply:
Oh, actually, I just discovered web3 has [`web3.utils.isAddress`](https://web3js.readthedocs.io/en/1.0/web3-utils.html#isaddress), which seems to do exactly what we need.

Reply:
We don't always have access to Web3 though, and passing it around a bunch of utility functions seems kinda meh...

Weirdly, you can't seem to use Web3.utils without initializing a provider first, so "offline" functions like this are hard to import directly >.<

Reply:
@c-johnson There's one trick to get at those bad boys that's used in `lib/txn.js'.  Agreed that it's weird, though.. :-)

``` js
const dummy = new Web3()
const toHex = dummy.utils.toHex
const toWei = dummy.utils.toWei
const fromWei = dummy.utils.fromWei
```

Reply:
Oh, oof, you're right. That sucks almost as much as reinventing the wheel.

Would it be weird to suggest a `/lib/ethUtils.js` that uses @jtobin's suggested method to expose "web3-less web3 functions"?

Reply:
Oh, interesting. It looks like the latest beta of Web3 actually breaks this; see this comment here (https://github.com/ethereum/web3.js/issues/2573#issuecomment-476231814). 

I didn't catch this during my beta.52 testing, but looks like this does actually break existing functionality. My love for the web3 library grows every day.

Looks like recommended usage is to just install web3-utils, which works fine with beta.52. Doing that now.

Reply:
This PR should address it; when it gets merged, I'll swap out our implementation for web3's isAddress

https://github.com/urbit/bridge/pull/112